File: backend/tiggle-root/tiggle/src/main/java/com/side/tiggle/domain/reaction/dto/ReactionDto.java
package com.side.tiggle.domain.reaction.dto;

import com.side.tiggle.domain.member.model.Member;
import com.side.tiggle.domain.reaction.model.Reaction;
import com.side.tiggle.domain.reaction.model.ReactionType;
import com.side.tiggle.domain.transaction.model.Transaction;
import lombok.*;
import lombok.experimental.SuperBuilder;

@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@SuperBuilder
@AllArgsConstructor
public class ReactionDto {
    private Long txId;
    private Long senderId;
    private Long receiverId;
    private ReactionType type;

    public Reaction toEntity(Transaction tx, Member sender, Member receiver) {
        return Reaction.builder()
                .tx(tx)
                .sender(sender)
                .receiver(receiver)
                .type(type)
                .build();
    }

    public static ReactionDto fromEntity(Reaction reaction) {
        return ReactionDto.builder()
                .txId(reaction.getTx().getId())
                .senderId(reaction.getSender().getId())
                .receiverId(reaction.getReceiver().getId())
                .type(reaction.getType())
                .build();
    }
}

File: backend/tiggle-root/tiggle/src/main/kotlin/com/side/tiggle/domain/reaction/dto/ReactionDto.kt
package com.side.tiggle.domain.reaction.dto

import com.side.tiggle.domain.reaction.model.Reaction
import com.side.tiggle.domain.reaction.model.ReactionType

open class ReactionDto(
    val txId: Long,
    val senderId: Long,
    val receiverId: Long,
    val type: ReactionType
) {

    companion object {
        fun fromEntity(reaction: Reaction): ReactionDto {
            return ReactionDto(
                txId = reaction.tx.id,
                senderId = reaction.sender.id,
                receiverId = reaction.receiver.id,
                type = reaction.type
            )
        }
    }
}